src/ir/hlir_types.rs

```diff
@@ -0,0 +1,21 @@
+use crate::SmlStr;
+
+enum NixPrimitive {
+    Bool(bool),
+    Int(i64),
+    Str(SmlStr),
+}
+
+trait NixValue {}
```

Comment:
why not model all possible values here (including rather opaque stuff like function calls)? Maybe it would be a good idea to put this into a separate crate.